network/manager.go

```diff
@@ -517,9 +533,17 @@ func (nm *networkManager) GetEndpointInfo(networkID, endpointID string) (*Endpoi
 
 	if nm.IsStatelessCNIMode() {
 		logger.Info("calling cns getEndpoint API")
-		epInfo, err := nm.GetEndpointState(networkID, endpointID)
+		epInfos, err := nm.GetEndpointState(networkID, endpointID)
+		if err != nil {
+			return nil, err
+		}
+		for _, epInfo := range epInfos {
```

Comment:
why return only for infra nic?

Reply:
I believe we discussed this before? https://github.com/Azure/azure-container-networking/pull/2688#discussion_r1609180201 We can only return 1 endpoint info from this method. Is there a different intended behavior/a case where we would want to return an endpoint info of type delegated vmnic?